### ISSUE TITLE
format: :short and format: :currency_short both fail

### ISSUE THREAD

**Issue by DaTrader** [kipcole9/money#135]
Is the below a bug or am I missing something:

```
iex(26)> Money.new( :usd, 1_000_000, format: :currency_short) |> to_string()
** (ArithmeticError) bad argument in arithmetic expression
    (ex_cldr_numbers 2.23.2) lib/cldr/number/formatter/short_formatter.ex:251: Cldr.Number.Formatter.Short.adjustment/2
    (ex_cldr_numbers 2.23.2) lib/cldr/number/formatter/short_formatter.ex:232: Cldr.Number.Formatter.Short.normalise_number/3
    (ex_cldr_numbers 2.23.2) lib/cldr/number/formatter/short_formatter.ex:175: Cldr.Number.Formatter.Short.choose_short_format/4
    (ex_cldr_numbers 2.23.2) lib/cldr/number/formatter/short_formatter.ex:95: Cldr.Number.Formatter.Short.short_format_string/6
    (ex_cldr_numbers 2.23.2) lib/cldr/number.ex:390: Cldr.Number.to_string/3
    (ex_money 5.7.1) lib/money.ex:773: Money.to_string!/2

iex(28)> Money.to_string( Money.new( :USD, 1_000_000), format: :currency_short)
** (ArithmeticError) bad argument in arithmetic expression
    (ex_cldr_numbers 2.23.2) lib/cldr/number/formatter/short_formatter.ex:251: Cldr.Number.Formatter.Short.adjustment/2
    (ex_cldr_numbers 2.23.2) lib/cldr/number/formatter/short_formatter.ex:232: Cldr.Number.Formatter.Short.normalise_number/3
    (ex_cldr_numbers 2.23.2) lib/cldr/number/formatter/short_formatter.ex:175: Cldr.Number.Formatter.Short.choose_short_format/4
    (ex_cldr_numbers 2.23.2) lib/cldr/number/formatter/short_formatter.ex:95: Cldr.Number.Formatter.Short.short_format_string/6
    (ex_cldr_numbers 2.23.2) lib/cldr/number.ex:390: Cldr.Number.to_string/3
```

While this works:

```
iex(30)> MyApp.Cldr.Number.to_string( 1_000_000, format: :currency_short)
{:ok, "$1M"}
```

In addition, why the following:

```
iex(33)> MyApp.Cldr.Number.to_string( 1_200_000, format: :currency_short)
{:ok, "$1M"}
```

does not produce `{ :ok, "$1.2M"}` instead?

Thanks,

Damir

**Comment by kipcole9** [comment]
Fixed in this [commit](https://github.com/elixir-cldr/cldr_numbers/commit/47e5ba4fc7ed2bc0cc85d60e91ed39b684601370) in [ex_cldr_numbers](https://github.com/ex_cldr_numbers). 

I have published [ex_cldr_numbers version 2.23.3](https://hex.pm/packages/ex_cldr_numbers/2.23.3) with the following changelog entry:

### Bug Fixes

* Fixes short formatting for decimal numbers. Thanks to @datrader for the report.  Closes #135 on [ex_money](https://github.com/kipcole9/ex_money).

**Comment by kipcole9** [comment]
On the issue of precision when short formatting, all of the short currency formats defined by ex_cldr are defined with no fractional digits. However that can be overridden with the `:fractional_digits` option. For example:
```elixir
iex> MyApp.Cldr.Number.to_string( 1_200_000, format: :currency_short)
{:ok, "$1M"}
iex> MyApp.Cldr.Number.to_string( 1_200_000, format: :currency_short, fractional_digits: 1)
{:ok, "$1.2M"}
iex> MyApp.Cldr.Number.to_string( 1_200_000, format: :currency_short, fractional_digits: 2)
{:ok, "$1.20M"}
```
